packages/discord.js/src/managers/GuildMemberManager.js

```diff
@@ -246,46 +248,65 @@ class GuildMemberManager extends CachedManager {
     const query = initialQuery ?? (users ? undefined : '');
 
     return new Promise((resolve, reject) => {
-      this.guild.client.ws.send(this.guild.shardId, {
-        op: GatewayOpcodes.RequestGuildMembers,
-        // eslint-disable-next-line id-length
-        d: {
-          guild_id: this.guild.id,
-          presences,
-          user_ids: users,
-          query,
-          nonce,
-          limit,
-        },
-      });
       const fetchedMembers = new Collection();
       let index = 0;
+
+      const cleanup = () => {
+        /* eslint-disable no-use-before-define */
+        clearTimeout(timeout);
+
+        this.client.ws.off(WebSocketShardEvents.Dispatch, rateLimitHandler);
```

Comment:
```suggestion
        this.client.ws.removeListener(WebSocketShardEvents.Dispatch, rateLimitHandler);
```
Consistency I guess...